### ISSUE TITLE
Timeout in calls to comparatively low energy consuming contracts

### ISSUE THREAD

**Issue by raidshift** [tronprotocol/java-tron#4311]
### 1. What did you do?
* I would like to create a clone of Tornado Cash (https://tornado.cash/) on TRON  
* Therefore I needed to deploy a ZK SNARK Verifier Contract (which performs some calls to precompiled ECDSA contracts) on TRON Blockchain:
https://tronscan.io/#/contract/TAvhRwrobjBsNLpKpTTsNfSVMAkELzuufB

### 2. What did you expect to see?
* _function verifyProof(bytes memory proof, uint256[6] memory input) public view returns (bool)_
to be executed without errors like in any other EVM compatible Network

* Test Parameters (for a simple test which should return the value _false_):
**proof** = 0x0c3f55fb97b02d2c507f8c5b6b87b8f8872939f07959f0f671bd83c767d42ca52e718ff1621ebe51cc7c831d39f5a3da25d1cd7db0b4d9e49d216762a3385f14273b6f3daf5b6f3e3ed77c0508117f3d651ddd6e8ac4f2b350dfca929d005504132acfbef922dff2af5a98e4f0d2115fbd8af99a30ebe18420b85ee2961b6dc905a114775273fcccfea094fa06ca9e0325b71bae23c6830f0a2c10737a5543f52072054a099fe527a0941005a44e8d0b055aa4065df5d90bd3966f242e5ad2ec0a9227d4ecdf1fbbfde91b96110e2259fba35fda009e3eaf2126af0d4d242c912e70a7083394fee6a6bc05ebd0d78f45621caa1053bbd2c1f0efde16e90cc59f
**input** = ["0","0","0","0","0","0"]

### 3. What did you see instead?

* When executing via [TronScan](https://tronscan.io/#/contract/TAvhRwrobjBsNLpKpTTsNfSVMAkELzuufB):
class org.tron.core.vm.program.Program$OutOfTimeException : CPU timeout for 'SWAP1' operation executing

* When executing via [EtherScan](https://etherscan.io/address/0xce172ce1f20ec0b3728c9965470eaf994a03557a#readContract):
a valid response: _false_

To sum it up it seems that it is not possible to call this kind of contracts (even if there are calls to low energy cosuming precompiled ECDSA contracts within the contract) due to harsh timeout restrictions on TRON

**Comment by vivian-kang** [comment]
Currently, the execution time of calling smart contract transactions cannot exceed 80ms. This 80ms parameter can be modified by SR voting.
So it seems that the contract code is too complex, so I will suggest to simplify the code logic and redeploy, Thanks.
Please refer to the guide: https://developers.tron.network/docs/faq#2-why-would-i-trigger-out_of_time-error-when-calling-contract-function

**Comment by yanghang8612** [comment]
According to the parameters you provided, we have done some tests. It takes about 120ms and 233,806 energy to complete your transaction execution.

But right now, TVM has a maximum execution time limit while executing a smart contract transaction, which is currently set at 80ms on the mainnet.
This is due to the fact that the goal of the TRON network when it was first launched was to provide high TPS and high stability. TRON network generates block every 3 seconds. When TVM was enabled for TRON network, we introduced a maximum execution time limit of TVM in order to maintain the stability of the network and to protect system contract type transactions.

There is a discussion about adjusting the maximum execution time limit of TVM. If you are interested, you are more than welcome to join. Here is the link [issue-384](https://github.com/tronprotocol/tips/issues/384).

**Comment by ethan1844** [comment]
Thanks for your contribution to java-tron, this issue will be closed because it was moved to [issue-384](https://github.com/tronprotocol/tips/issues/384).

Please feel free to re-open it if you still see the issue, thanks.
